### ISSUE TITLE
[SSI] Client can't immediately reconnect when connection times out

### ISSUE THREAD

**Issue by slithy** [xrootd/xrootd#776]
The client behaves differently in the following situations:

### 1. Server is stopped and restarted immediately
- Client gets `[ERROR] Session invalid` on next attempt to send a Request
- A new XrdSsiService object is created
- Client can reconnect immediately to the new server instance

### 2. Server is stopped and restarted after some time
- Client gets `[ERROR] Operation expired` on next attempt to send a Request
- A new XrdSsiService object is created
- The new object cannot be used to connect, it continues to give `[ERROR] Operation expired`
- After several attempts at recreating XrdSsiService object, eventually it succeeds (after ~60 seconds)
- Client can reconnect to the new server instance

In the second case, it should be possible to reconnect as soon as the server is running.

**Comment by slithy** [comment]
In ticket #775 you mentioned that the 60 second delay between connection attempts is controlled by the environment variable `XRD_CONNECTIONWINDOW`.

This variable controls the length of time allowed to establish a connection. This is a different concept from the length of time that must elapse after a failure before the connection can be retried. The desired behaviour is that after a failure, the client should be able to retry immediately with zero delay.

In other words, the behaviour after `Session invalid` and `Operation expired` should be the same.

**Comment by simonmichal** [comment]
Just to clarify, the ConnectionWindow is the time the client has been given to establish the TCP connection and carry out the XRootD hand shake. However, if the client fails to connect to the server and there are no more addresses to try, it will retry only by the end of the current ConnectionWindow (hence it does have an impact on the length of time before a retry).

**Comment by xrootd-dev** [comment]
Hi Michael,

The quick answer is we wanted to prevent a reconnection storm while a 
server was getting back on its feet.

Now, back to your problem. I am trying to get a quick bypass while we work 
on a long term solution. You can adjust the connection window without 
setting an envar (though an envar doesn't require you changing any code) 
by calling XrdSsiProvider::SetTimeout(connect_T, <tvalue>). So, at this 
point you can set the connection timeout window to be say 5 to 10 seconds. 
I admit, I don't know how quickly quests pile up and that may not solve 
the problem in all cases.

Andy

**Comment by abh3** [comment]
This issue has been resolved.
